middleware/cacheHandler.js

```diff
@@ -0,0 +1,61 @@
+const config = require('config')
+const urlGetter = require('url')
+
+const getUrl = (req) => {
+  return urlGetter.format({
+    protocol: req.protocol,
+    host: req.get('host'),
+    pathname: req.originalUrl
+  })
+}
+
+class CachedData {
+  constructor (requestUrl, timeStamp, data) {
+    this.url = requestUrl
+    this.timeStamp = timeStamp
+    this.data = data
+  }
+
+  getData () {
+    return this.data
+  }
+
+  isExpired () {
+    return this.timeStamp + cacheMillisecondsToLive < Date.now()
+  }
+}
+
+const cacheDataMap = new Map()
+const cacheMinutesToLive = 10
+const cacheMillisecondsToLive = cacheMinutesToLive * 60 * 1000
+
+const saveToCache = (url, data) => {
+  cacheDataMap.set(url, new CachedData(url, Date.now(), data))
```

Comment:
This design will make it harder for you to transition to Redis, since Redis is a key value DB. the map is a good idea but the transition will be easier for you if the value will be a plain json and not a class.
anyway, we can change it now or later, up to you. not urgent.